### PR TITLE
Rename the 'Melange Playground' chapter to 'Numeric Types' and add more detail

### PR DESCRIPTION
**Pull request by feihong**
Also divide the Counter chapter into more sections.

### CODE REVIEW

File: docs/numeric-types/index.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+```reason
+let foo: int = 42;
+let bar: float = 42.1;
+Js.log(foo);
+Js.log(bar);
+```
+
+## Melange Playground
+
+[Melange Playground](https://melange.re/v2.0.0/playground/) is an interactive
```
Comment (jchavarri) [inline docs/numeric-types/index.md]: This is a nit, but we can use links to v2.1.0 now (applies to other links in the PR)

File: docs/counter/index.md [hunk 2]
```diff
@@ -46,36 +45,47 @@ to decorate `make` with the
...
+them to the `React.element` type. This is exactly what the `React.string`
+function does---if you hover over it, you'll see that it displays the type
+`string => React.element`.
+
+## Using the App component
 
 A little bit further down, we make use of the `App` component:
 
-<<< ./Snippets.re#use-app-component
+<<< Snippets.re#use-app-component
```
Comment (jchavarri) [inline docs/counter/index.md:59]: Why was the `./` dropped? Just not necessary?

File: docs/numeric-types/Snippets.re [hunk 3]
```diff
@@ -0,0 +1,75 @@
+// #region counter
+module Counter = {
+  [@react.component]
+  let make = () => {
+    let (counter, setCounter) = React.useState(() => 0);
+
+    <div
+      style={ReactDOMStyle.make(
```
Comment (jchavarri) [inline docs/numeric-types/Snippets.re:8]: ftr this is changing soon to `ReactDOM.Style` in reasonreact. Maybe worth opening an issue to remember or something.

**Review by jchavarri — APPROVED** [review]
This looks great, thank you! ❤️

File: docs/counter/index.md (hunk 2 above)
Comment (feihong) [reply docs/counter/index.md:59]: Yeah, when I first added it I assumed it was needed, but it's not

File: docs/numeric-types/Snippets.re (hunk 3 above)
Comment (feihong) [reply docs/numeric-types/Snippets.re:8]: Create new issue: https://github.com/melange-re/melange-for-react-devs/issues/3#issue-1977099173

File: docs/numeric-types/index.md (hunk 1 above)
Comment (feihong) [reply docs/numeric-types/index.md]: Updated links in https://github.com/melange-re/melange-for-react-devs/pull/2/commits/99e33ffbe4ae012c9300e563738d62239cc0bb7e

**Pull request merged**